jgiven-core/src/main/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportBlockConverter.java

```diff
@@ -0,0 +1,432 @@
+    }
+
+    private static String buildDataTableHead(final DataTable dataTable) {
+        final String colSpec = generateTableColSpec(dataTable.hasVerticalHeader(), dataTable.getColumnCount());
+
+        return "[.jg-argumentTable"
+            + (dataTable.hasHorizontalHeader() ? "%header" : "")
+            + ",cols=\"" + colSpec + "\"]";
+    }
+
+    private static String buildParameterWordFragment(final String placeHolderValue) {
+        return " [.jg-argument]*<" + placeHolderValue + ">*";
+    }
+
+    private static String buildArgumentWordFragment(final String argumentValue) {
+        if (argumentValue.contains(NEW_LINE)) {
+            return NEW_LINE
+                + "+" + NEW_LINE
+                + "[.jg-argument]" + NEW_LINE
+                + "...." + NEW_LINE
+                + argumentValue + NEW_LINE
+                + "...." + NEW_LINE;
+        } else {
+            return " [.jg-argument]_" + escapeArgumentValue(argumentValue) + "_";
+        }
+    }
+
+    private static String buildOtherWordFragment(final String word, final boolean differs) {
+        if (differs) {
+            return " #" + word + "#";
+        } else {
+            return " " + word;
+        }
+    }
+
+    private static String buildStepEndFragment(final int depth, final boolean caseIsUnsuccessful,
+                                               final StepStatus status,
+                                               final long duration,
+                                               final String extendedDescription) {
+        final String humanDuration = toHumanReadableDuration(duration).map(dur -> " (" + dur + ")").orElse("");
+        final String stepStatus = caseIsUnsuccessful
+            ? " " + toHumanReadableStatus(status) + humanDuration
+            : "";
+
+
+        if (extendedDescription != null && !extendedDescription.isEmpty()) {
+            return stepStatus + " +" + NEW_LINE
+                + buildIndentationFragment(depth, " ") + " _+++" + extendedDescription + "+++_";
+        } else {
+            return stepStatus;
+        }
+    }
+
+    private static String escapeTableValue(final String value) {
+        return escapeArgumentValue(value.replace("|", "\\|"));
+    }
+
+    private static String escapeArgumentValue(final String value) {
+        // TODO Is this really necessary?
+        // return "+" + value + "+";
+        return value;
+    }
+
+    private static String buildIndentationFragment(final int depth, final String symbol) {
+        return generate(() -> symbol).limit(depth + 1L).collect(joining());
+    }
+
+    private static String generateTableColSpec(final boolean withVerticalHeader, final int columnCount) {
+        return withVerticalHeader
+            ? "h," + generate(() -> "1").limit(columnCount - 1L).collect(joining(","))
+            : generate(() -> "1").limit(columnCount).collect(joining(","));
+    }
+
+    private static String toAsciiDocTag(final ExecutionStatus executionStatus) {
```

Comment:
How about putting these enum conversion methods in a package-private `EnumValuesMappingUtil` so that this class does not need to concern itsself with the mapping details?